contracts/tokenomics/generator/src/contract.rs

```diff
@@ -166,7 +173,32 @@ pub fn execute(
             }
 
             for (addr, alloc_point) in pools {
-                setup_pools.push((addr_validate_to_lower(deps.api, &addr)?, alloc_point));
+                let pool_addr = addr_validate_to_lower(deps.api, &addr)?;
+                let pair_info = pair_info_by_pool(deps.as_ref(), pool_addr.clone())?;
+
+                // check if assets in the blocked list
+                let mut is_blocked = false;
+                for asset in pair_info.asset_infos.clone() {
+                    if cfg.blocked_list_tokens.contains(&asset) {
+                        is_blocked = true;
+                        break;
+                    }
+                }
+
+                // If a pair gets deregistered from the factory, no one can set its generator
+                // alloc_points to anything above 0.
+                let resp: StdResult<PairInfo> = deps.querier.query_wasm_smart(
+                    cfg.factory.clone(),
+                    &FactoryQueryMsg::Pair {
+                        asset_infos: pair_info.asset_infos,
+                    },
+                );
+
+                if is_blocked || resp.is_err() {
+                    setup_pools.push((pool_addr, Uint64::zero()));
```

Comment:
We should raise an error and not set alloc_point to zero implicitly. It will help to test the generator controller works properly. Also let's add possibility to query blocklisted tokens by the generator controller and others

Reply:
When it is blocked too

Reply:
When it is blocked too